compiler/rustc_mir_transform/src/instsimplify.rs

```diff
@@ -252,6 +255,28 @@ impl<'tcx> InstSimplifyContext<'tcx, '_> {
         terminator.kind = TerminatorKind::Goto { target: destination_block };
     }
 
+    fn simplify_nounwind_call(&self, terminator: &mut Terminator<'tcx>) {
```

Comment:
Where is this coming from?

Reply:
The `precondition_check` function is a nounwind function, but without this optimization we sometimes generate landing pads out of it.

Reply:
"sometimes"? Strange. Is this a MIR building bug?

Reply:
I don't know. I tried to fix this in MIR building and it seemed really awkward.

I also wonder if inlining will cause the optimization to sometimes see a monomorphic call that was initially built polymorphic. And thus we'd need the opt anyway.

Reply:
Ah, inlining is a good point. Might be worth a dedicated test.